src/main/java/com/ita/if103java/ims/service/impl/WarehouseServiceImpl.java

```diff
@@ -103,10 +105,10 @@ private WarehouseDto createNewWarehouse(WarehouseDto warehouseDto, UserDetailsIm
                 AddressDto addressDto = addressDtoMapper.toDto(addressDao.findByWarehouseId(warehouse.getId()));
                 warehouseDto.setAddressDto(addressDto);
             }
-
         }
+        List<WarehouseDto> warehouses = warehouseDtoMapper.toDtoList(all);
+        return new PageImpl<>(warehouses, pageable, all.size());
```

Comment:
all.size() will not work here. Need to call DB to count total warehouses in the DB per Account. Because all.size() is the amount less then or equal to pageSize provided in the pageable and not the total amount of the records.